src/js/background.js

```diff
@@ -235,29 +235,70 @@ Badger.prototype = {
   * add any new entries that already have a parent domain in the action_map
   * and remove any old entries that are no longer in the cookie block list
   * from the action map
+  *
+  * @param {Function} [callback] optional callback, gets success status boolean
   **/
-  updateCookieBlockList: function(){
+  updateCookieBlockList: function (callback) {
     var self = this;
-    utils.xhrRequest(constants.COOKIE_BLOCK_LIST_URL, function(err,response){
-      if(err){
-        console.error('Problem fetching cookieblock list at',
-                  constants.COOKIE_BLOCK_LIST_URL, err.status, err.message);
-        return;
+
+    if (!callback) {
+      callback = _.noop;
+    }
+
+    utils.xhrRequest(constants.COOKIE_BLOCK_LIST_URL, function (err, response) {
+      if (err) {
+        console.error(
+          "Problem fetching cookieblock list at",
+          constants.COOKIE_BLOCK_LIST_URL,
+          err.status,
+          err.message
+        );
+
+        return callback(false);
       }
-      var cookieblock_list = self.storage.getBadgerStorageObject('cookieblock_list');
-      var action_map = self.storage.getBadgerStorageObject('action_map');
 
-      var newCbDomains = _.map(response.split("\n"), function(d){ return d.trim();});
+      // handle empty response
+      if (!response.trim()) {
+        return callback(false);
+      }
+
+      var newCbDomains = _.map(response.trim().split("\n"), domain => domain.trim());
+
+      // validate the response
+      if (!_.every(newCbDomains, (domain) => {
+        // all domains must contain at least one dot
+        if (domain.indexOf('.') == -1) {
+          return false;
+        }
+
+        // validate character set
+        // TODO overly restrictive but that's OK for now, we manage the list
```

Comment:
can you add a comment for those not regex inclined? 
`// starts with (a through z or 0 through 9) then has one or more (a through z or 0 through 9 or "." or "-") and ends with (a through z)`

I think that is right.